### DISCUSSION TITLE
Text Editor's Status Bar

### DISCUSSION THREAD

**Discussion by joegittingherdone** [open-watcom/open-watcom-v2#773] [Q&A] [answered]
I bought the Watcom compiler for Windows in the 1990s and thought that the status bar at the bottom of the Watcom Text Editor worked differently then, as compared to v2 today.  So I'm trying to understand how it works now.  When modifying the source code (changing "Line:" to "line:" in weditor.ini for example), it does not change in the bottom-most status bar within viw.exe.  Does anyone know where the text for it is stored before rendering (like "Line:", "Col:", "Mode:", etc.)?

**Comment by jmalak** [comment]
In OpenWatcom version of VI editor is status line configurable by **statusstring** line in configuration file.
Syntax description is in VI editor guide [statusstring](https://open-watcom.github.io/open-watcom-v2-wikidocs/vi.html#statusstring).

What is probably new is binding configuration file into executable, see [EDBIND utility](https://open-watcom.github.io/open-watcom-v2-wikidocs/vi.html#Using_Edbind). The important is if you change anything in configuration file you must re-bind appropriate file into VI executable to be permanent change.

Next you must delete your own local copy of configuration file (where your setup changes are) from your profile to be new configuration file version loaded from VI executable.
Unfortunately you can change your own local copy of configuration file in your profile directly, but if you use VI editor on the other PC you will need to modify it again. If you re-bind changed configuration file to executable by edbind then if you use this executable anywhere it will automaticaly use your setup.

Below is list of all **"binded"** files to executable (much more then in old **"Watcom"** version).
These files are precompiled [see COMPILE command](https://open-watcom.github.io/open-watcom-v2-wikidocs/vi.html#COMPILE) to save space in executable (including configuration file).

VI GUI version list of binded files:
weditor.ini
wrme.vi
rdmewin.vi
rcs.vi
menuwin.vi
mlselw.vi
mcselw.vi
untitled.vi
writewin.vi
writeall.vi
qall.vi
proc.vi
err.vi
chkout.vi
forceout.vi
unlock.vi
fin.vi
toggle.vi
ideact.vi
ddesinit.vi
ddecb.vi
error.dat
keys.dat
awk.dat
basic.dat
bat.dat
c.dat
cpp.dat
dbtest.dat
declspec.dat
fortran.dat
gml.dat
html.dat
java.dat
mif.dat
perl.dat
pragma.dat
rc.dat
sql.dat
wml.dat

**Reply by joegittingherdone** [reply]
I think I understand.  Does this mean there is no way for me to alter the way the status bar renders (via weditor.ini) for someone who downloads the Open Watcom end-user installer from GitHub in the future because he will have to recompile my weditor.ini first and then bind it to his final viw.exe himself?  

As displays have gained resolution over time, I think the status bar has become illegible so I was looking to repair it:

![stat-bar](https://user-images.githubusercontent.com/16060716/144731606-a96f6eb9-8272-49a5-a324-d7ef22e652ed.jpg)

**Comment by jmalak** [comment, marked as answer]
You can change it in your profile in weditor.ini. by example on my Windows 10 it is located in 
C:\Users\<user name>\AppData\Roaming\Open Watcom\weditor.ini
But this change is your user local change.
Anyway what you show is related to the font change.
On my system I have no such problem.
I use default status bar setup
statussections = 78 137 250 317 
statusstring = "Line:$5L$[Col:$3C$[Mode: $M$[$|$T$[$H"
statuswindow
    dimension 0 23 7 24
    border -1
    text 0 7 17
    hilight 15 0 0
endwindow
to reconfigure it from GUI you can press F9 and move field from pallet and put it on your preferred position
or you can change statussections manually to what you preferred.
![image](https://user-images.githubusercontent.com/2017260/144737045-8490537c-1a97-4ffa-9e9a-59268a56f2ea.png)
Could you give me a copy of your configuration file in
C:\Users\<user name>\AppData\Roaming\Open Watcom\weditor.ini
Description of parameters is in vi documentation
[statussections](https://open-watcom.github.io/open-watcom-v2-wikidocs/vi.html#statussections)
[statusstring](https://open-watcom.github.io/open-watcom-v2-wikidocs/vi.html#statusstring)
[statuswindow](https://open-watcom.github.io/open-watcom-v2-wikidocs/vi.html#STATUSWINDOW)

**Comment by jmalak** [comment]
Sorry if I confused you.
The bind version of weditor.ini is default setup "template" which is used for start VI editor if you don't have your own in your profile.
It is used until you create your own copy by "Save Configuration". As soon as your own copy of weditor.ini exists then vi editor use it instead of default "template".
If you remove your own copy of weditor.ini it will use again this default template.
You can simply pass weditor.ini between PCs, but take into account that it hold geometry and position in windows coordinates which depend on Windows resolution setup.
If you have any trouble with vi configuration then you can delete your own weditor.ini file and vi will start with default setup from bind version.
For portable change of vi configuration is better to use vi script which can be run in vi editor and do only appropriate changes in weditor.ini for your purpose after OW vi editor installation instead of replacement of full weditor.ini file with potential side effects.

**Comment by joegittingherdone** [comment]
I think I understand.  The following is what is in my weditor.ini; I don't remember changing it before.

set statusstring = "Line:$5L$[Col:$3C$[Mode: $M$[$|$T$[$H"
set statussections = 60 105 192 244 

Setting these to your numbers made it better.  My "overstrike" doesn't fit in the section after I press the Insert key on the keyboard.

![stat-bar-ovrstrk](https://user-images.githubusercontent.com/16060716/144774489-cdd2c39d-6ffe-4b8c-8d11-29b7ae9849a0.jpg)

I noticed the F9 dialog box does not allow dragging components to the status bar; it shows me the "can't do that" symbol: the circle with a diagonal bar going across its diameter.  Might you know why that is?

**Comment by jmalak** [comment]
It looks like re-sizing of section size by mouse doesn't work, but I can drag and drop fields from F9 palette to status bar and rearrange it.
You can do section on status bar wider to be correct for your font size by manual change of staussections values.
I will look on the problem with status bar section sizes (probably bug in vi editor).

this is for statussections = 78 137 250 317 
![image](https://user-images.githubusercontent.com/2017260/144799741-ca9522ee-b68b-4b51-8454-867c58f51afd.png)
and this is for statussections = 78 160 400 500
![image](https://user-images.githubusercontent.com/2017260/144800203-a1b3cb21-b6c0-4258-9a8d-8bf63d45fdb0.png)
it define start of each section, first start always on begining of status bar and 5 sections are defined by default.
There is separated guide for GUI version of Vi [The Open Watcom Editor](https://open-watcom.github.io/open-watcom-v2-wikidocs/guitool.html#The_Open_Watcom_Editor) and [Configuration of status bar](https://open-watcom.github.io/open-watcom-v2-wikidocs/guitool.html#Configuring_the_Editor)

**Comment by jmalak** [comment]
I checked re-sizing of status bar sections on 16-bit Windows version and it works.
It must be fixed for 32/64-bit Windows. It looks like the code for 32/64-bit Windows need to be different then for 16-bit Windows (border of section is wider the for 32/64-bit Windows vertical bar only).

**Reply by joegittingherdone** [reply]
I cannot drag from the F9 box to the status bar in my current Open Watcom installation, but it is an older installation.  I CAN drag from the F9 box after compiling viw.exe from source code today.  Thank you for your help.

**Comment by jmalak** [comment]
Anyway in VI for 32/64-bit Windows is bug, because you cannot resize each section by mouse by moving vertical bar between sections.
It works only for 16-bit Windows. Original VI from Watcom has this working also for 32-bit Windows.
You cannot rebuild OW from sources you can download [latest CI-build](https://github.com/open-watcom/open-watcom-v2/releases/download/Last-CI-build/ow-snapshot.tar.gz) and copy it somewhere for testing.

**Reply by joegittingherdone** [reply]
While we're on the topic, I noticed that [this](https://open-watcom.github.io/open-watcom-v2-wikidocs/vi.html#statusstring) says: 

> For GUI editors, the default status string setting is:     
>      set statusstring = "Ln:$5L$[Col:$3C$[Mode: $M$[$|$T$[$H"

which is incorrect (it should read "Line:$5"), but I don't how to change that documentation.

**Reply by jmalak** [reply]
I don't understand what is wrong with

> set statusstring = "Ln:$5L$[Col:$3C$[Mode: $M$[$|$T$[$H"

Anyway documentation is writen in [WGML language (Markup)](https://open-watcom.github.io/open-watcom-v2-wikidocs/wgmlref.pdf) and [short description of documentation handling](https://raw.githubusercontent.com/open-watcom/open-watcom-v2/master/docs/howto.txt).
This part is in [source file](https://github.com/open-watcom/open-watcom-v2/blob/master/docs/doc/vi/settings.gml) .

**Reply by joegittingherdone** [reply]
> For GUI editors, the default status string setting is:
> set statusstring = "Ln:$5L$[Col:$3C$[Mode: $M$[$|$T$[$H"

The graphics that you and I attached above clearly say "Line", not "Ln".  It's incorrect, but I don't know if it's worth changing; you can change it if you want.

**Reply by jmalak** [reply]
Oh, I see. 
Thanks for your  info.
Documentation is fixed now.
